File: model/src/main/java/com/sreMake/model/system/RoleResources.java
package com.sreMake.model.system;

import com.sreMake.model.BaseEntity;
import com.sreMake.model.user.Role;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Key;
import org.babyfish.jimmer.sql.ManyToOne;
import org.babyfish.jimmer.sql.Table;

@Entity
@Table(name = "srm_role_resources")
public interface RoleResources extends BaseEntity {
    @ManyToOne
    @Key
    Role role();

    @ManyToOne
    @Key
    Resources resources();
}

File: model/src/main/kotlin/com/SReMake/model/system/RoleResources.kt
package com.SReMake.model.system

import com.SReMake.model.BaseEntity
import com.SReMake.model.user.Role
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Key
import org.babyfish.jimmer.sql.ManyToOne
import org.babyfish.jimmer.sql.Table

@Entity
@Table(name = "srm_role_resources")
interface RoleResources : BaseEntity {
    @ManyToOne
    @Key
    val role: Role

    @ManyToOne
    @Key
    val resource: Resources
}